docs/general/session.md

```diff
@@ -35,4 +35,41 @@ backends can link the two sessions.
 <!-- END AUTOGENERATED TEXT -->
 <!-- endsemconv -->
 
+## Session Events
+
+### Session Start Event
+
+![Experimental](https://img.shields.io/badge/-experimental-blue)
+
+`event.name`=`session.start`
+
+For instrumentation that tracks user behavior during user sessions, a `session.start` event MUST be emitted
+every time a session is created. When a new session is created as a continuation of a prior session,
+the `session.previous_id` SHOULD be included in the event. The values of `session.id` and `session.previous_id`
+MUST be different.
+
+When the `session.start` event contains both `session.id` and `session.previous_id` fields, the event then implies
+that the previous session has ended. If the session ID in `session.previous_id` has not yet ended via explicit
+`session.end` event, then the receiver SHOULD treat this continuation event as semantically equivalent to
+`session.end(session.previous_id)` and `session.start(session.id)`.
+
+| Body field                                                    | Type   | Description                                          | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) | Stability                                                        |
+|---------------------------------------------------------------|--------|------------------------------------------------------|-------------------------------------------------------------------------------------------------------|------------------------------------------------------------------|
+| [`session.id`](/docs/attributes-registry/session.md)          | string | The ID of the new session being started.             | `Required`                                                                                            | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`session.previous_id`](/docs/attributes-registry/session.md) | string | The previous `session.id` for this user, when known. | `Conditionally Required`                                                                              | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+
+### Session End Event
+
+![Experimental](https://img.shields.io/badge/-experimental-blue)
+
+`event.name`=`session.end`
+
+For instrumentation that tracks user behavior during user sessions, a `session.end` event SHOULD be emitted
+every time a session ends. When a session ends and continues as a new session, this event SHOULD be
+emitted prior to the `session.start` event.
```

Comment:
since this event is optional, s/SHOULD/MAY/g

Reply:
I thought SHOULD was a bit more suggestive of "Oh hey this is a good idea you probably want to do this..." versus MAY which is like "Doesn't matter, do this if you want".

Reply:
For the browser case, there will be a scenario of session timing out when `session.end` event can be emitted. However, it's better to avoid this since it's not possible to emit the event in all the scenarios when session ends. To avoid confusion, it's better to not emit this event at all.  It is for this reason I suggest using MAY instead of SHOULD.

Reply:
> To avoid confusion, it's better to not emit this event at all.

While I definitely agree that there will be cases where the client instrumentation cannot send the `session.end`, I disagree that this should be discouraged. Backends that receive a session.end can begin processing a complete session immediately, instead of waiting for a much longer timeout. Prefer to keep it SHOULD.